### ISSUE TITLE
Vault MySQL secret backend credentials created but not renewable, Vault v0.3.1

### ISSUE THREAD

**Issue by ccampanale** [hashicorp/vault#905]
Vault MySQL secret backend credentials are created but not renewable using Vault v0.3.1. From the DB perspective these credentials are being created and Vault initially responds with `lease_renewable true` but the lease fails to be properly renewed. 

```bash
[root@consul1 ~]# vault read mysql/creds/readwrite
Key             Value
lease_id        mysql/creds/readwrite/0c0af69c-2754-98dc-0192-8990527519c5
lease_duration  3600
lease_renewable true
password        ccedf8d7-d5e2-3523-8cce-5964587c5504
username        root-6e6c6e9f-e9
[root@consul1 ~]#
[root@consul1 ~]# vault renew -format=json mysql/creds/readwrite/0c0af69c-2754-98dc-0192-8990527519c5
{
    "lease_id": "",
    "lease_duration": 0,
    "renewable": false,
    "data": null
}
```

**Configuration**

```bash
[root@consul1 ~]# vault mount -path=mysql mysql
Successfully mounted 'mysql' at 'mysql'!
[root@consul1 ~]#
[root@consul1 ~]# vault write mysql/config/connection  value='vault:vault@tcp(db-test.service.consul:3306)/'
Success! Data written to: mysql/config/connection
[root@consul1 ~]#
[root@consul1 ~]# vault write mysql/config/lease  lease='1h' lease_max='24h'
Success! Data written to: mysql/config/lease
[root@consul1 ~]# vault write mysql/roles/readwrite sql="CREATE USER '{{name}}'@'%' IDENTIFIED BY '{{password}}';GRANT ALL ON db_test.* TO '{{name}}'@'%';"
Success! Data written to: mysql/roles/readwrite
```

The configuration is really straight forward and it all seems to work great except for the renewal process. Revoking the lease also appears to work fine as it returns that the key was revoked and the credentials no longer appear in the DB.

Please let me know if there is anything I can test or get for you to review.

**Comment by jefferai** [comment]
I think what you are seeing there is an artifact of using `--format=json`: I think you're seeing fields that are not being set in the response and thus showing the defaults, which for a boolean is false -- something backed up by the fact that the `lease_id` is empty. Is the lease actually being revoked after the original expiration period or is it valid for the renewal period as well?

**Comment by ccampanale** [comment]
That would make sense but I do actually think the lease is being revoked. FYI - this is related to my CT issue (https://github.com/hashicorp/consul-template/issues/493). With the issue in CT I have noticed that the following output is generated in the logs:

```
2015/12/29 15:35:04 [WARN] ("vault(mysql/creds/readwrite)") lease duration is 0, setting to 5m
2015/12/29 15:35:04 [DEBUG] ("vault(mysql/creds/readwrite)") vault returned the secret
```

This leads me to believe that CT may also be seeing the lease_id of 0. This also makes me question if CT is trying to update the data but since the renew doesn't provide it, the output ends up having `<no value>`. 

The JSON output does seem likely as without it I get the following:

```bash
[root@consul1 ~]# vault renew mysql/creds/readwrite/0c0af69c-2754-98dc-0192-8990527519c5
Key	Value
```

But this does also make me question, should data be returned here?

I'll need to adjust the lease_max to something lower to be absolutely certain that they are/aren't renewing properly but I do know that CT seems to be creating new credentials every hour so my guess is that it isn't actually renewing. I'll test and be sure manually though using the CLI.

**Comment by jefferai** [comment]
I don't know much about CT's behavior but I'm not sure it does (or doesn't) do the right thing in all situations w.r.t. renewing vs simply generating new credentials.

As for the CLI, regardless of format, it picks and chooses which bits of a response to show, and generally may get it wrong for any individual response. The best way to see what's actually coming back is to use the JSON API directly, as any CLI command is simply filtering out data returned from that call.

**Comment by ccampanale** [comment]
Ahh good call, I'll also give that a try.

**Comment by ccampanale** [comment]
OK, I did some more testing using curl and it looks like the lack of data is normal (which confirms what you mentioned about the renewable field, etc.) and appears that the renewal is actually working (tested with smaller lease periods to be sure). 

So this is good to go, sorry about that! Thanks for helping me research this though!

I did find more information to add to the issue for CT though as it appears to be trying to renew the lease twice then immediately re-reading the secret which just seems strange... Thanks again!
